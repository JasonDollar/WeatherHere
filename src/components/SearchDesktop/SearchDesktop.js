import React from 'react'
import PropTypes from 'prop-types'
import Icon from '../Icon/Icon'
import classes from './SearchDesktop.module.scss'

const SearchDesktop = ({ 
  onSearchFormSubmit, onInputChange, inputValue, text, onSettingIconClick, getUserLocation,
}) => (
  <div className={classes.Search}>
    <form onSubmit={onSearchFormSubmit} className={classes.form}>
      <input type="text" onChange={onInputChange} value={inputValue} className={classes.input} placeholder={text.inputPlaceholder} />
      <button className={`${classes.button} ${classes.formButton}`} type="submit">
        <span className="hidden">{text.search}</span>
        <Icon icon="search" viewBox="0 0 50 50" title="" />
        <span>{text.search}</span>
          
      </button>
    </form>
    <button className={`${classes.button} ${classes.buttonGeo}`} onClick={getUserLocation} type="button">
      <span className="hidden">{text.geoFull}</span>
      <Icon icon="gpsArrow" viewBox="0 0 51.636 51.636" title="" />
      <span>{text.geo}</span>
        
    </button>
    <button className={`${classes.button} ${classes.buttonSettings}`} onClick={onSettingIconClick} type="button">
      <span className="hidden">{text.settings}</span>
      <Icon icon="settings" viewBox="0 0 50 50" title="" />
      <span>{text.settings}</span>
        
    </button>
  </div>

)

export default SearchDesktop

SearchDesktop.propTypes = {
  onSearchFormSubmit: PropTypes.func.isRequired, 
  onSettingIconClick: PropTypes.func.isRequired, 
  onInputChange: PropTypes.func.isRequired, 
  inputValue: PropTypes.string.isRequired, 
  text: PropTypes.shape({
    search: PropTypes.string,
    settings: PropTypes.string,
    forecast: PropTypes.string,
    geo: PropTypes.string,
    geoFull: PropTypes.string,
    inputPlaceholder: PropTypes.string,
    icon: PropTypes.string,
  }).isRequired,
  getUserLocation: PropTypes.func.isRequired, 
}