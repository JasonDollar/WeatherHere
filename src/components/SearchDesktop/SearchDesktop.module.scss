
.Search {
  flex: 1 1;
  display: flex;
  align-content: flex-end;
  padding-bottom: 1rem;
  margin: 0;

  @media only screen and (min-width: 806px) {
    padding: 1rem 0 1rem 1rem;
  }
}

.form {
  flex: 1;
  display: flex;
  justify-content: space-between;
  margin: 0 1rem;
  position: relative;

  & .input {
    flex: 1;    
    margin: 0rem auto;
    padding: .5rem .5rem;
    max-width: 500px;
    border: 1px solid transparent;
    border-bottom: 1px solid var(--color-text-transparent);
    border-radius: 0;
    font-size: 25px;
    color: var(--color-text-secondary);
    background: var(--color-primary);
    outline: none;

    &:focus {
      border-bottom: 1px solid var(--color-text-secondary);
    }

    &::placeholder {
      color: var(--color-text-transparent);
    }
  }

  & .formButton {
    & svg {
      height: var(--normal-text-font-size);
      width: var(--normal-text-font-size);
      fill: var(--color-text-secondary);
      transform: scale(1.5);
    }
  }
}

.button {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: none;
  border: 1px solid transparent;
  padding: .5rem 2rem 0 2rem;
  color: var(--color-text-secondary);
  cursor: pointer;

  & span {
    padding-top: .7rem;
    font-size: 14px;
  }
}

.buttonGeo {
  margin: 0 1rem;
  
  & svg {
    height: var(--normal-text-font-size);
    width: var(--normal-text-font-size);
    fill: var(--color-text-secondary);
    transform: scale(1.5);
  }
}

.buttonSettings {
  padding: 0.5rem 0 0;
  margin: 0 1rem;
  & svg {
    height: var(--normal-text-font-size);
    width: var(--normal-text-font-size);
    fill: var(--color-text-secondary);
    transform: scale(1.5);
  }
}