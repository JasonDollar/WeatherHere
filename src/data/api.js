const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

