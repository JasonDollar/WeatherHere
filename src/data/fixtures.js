import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}