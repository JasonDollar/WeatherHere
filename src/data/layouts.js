const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



