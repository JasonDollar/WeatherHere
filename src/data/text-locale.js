const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 