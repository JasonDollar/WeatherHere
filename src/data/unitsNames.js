export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}