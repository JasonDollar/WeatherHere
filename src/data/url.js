const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }