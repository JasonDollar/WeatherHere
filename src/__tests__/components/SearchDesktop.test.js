import React from 'react'
import { shallow } from 'enzyme'
import { appTestData as data } from '../../data/fixtures'
import SearchDesktop from '../../components/SearchDesktop/SearchDesktop'

const testProps = {
  onSearchFormSubmit: jest.fn(),
  onInputChange: jest.fn(),
  inputValue: '',
  text: data.text.pl.layout,
  getUserLocation: jest.fn(),
  onSettingIconClick: jest.fn(),
}
let wrapper

beforeAll(() => {
  wrapper = shallow(<SearchDesktop {...testProps} />)
})

describe('SearchDesktop component tests', () => {
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('simulate input change', () => {
    const payload = { target: 'City name' }
    const element = wrapper.find('input')
    element.simulate('change', payload)
    expect(testProps.onInputChange).toHaveBeenCalledTimes(1)
    expect(testProps.onInputChange).toHaveBeenLastCalledWith(payload)
  })

  test('geolocation button click', () => {
    const element = wrapper.find('.buttonGeo')
    element.simulate('click')
    expect(testProps.getUserLocation).toHaveBeenCalledTimes(1)
  })

  test('simulate form submition', () => {
    // const payload = { target: 'City name' }
    const element = wrapper.find('form')
    element.simulate('submit', { preventDefault: () => { } })
    expect(testProps.onSearchFormSubmit).toHaveBeenCalledTimes(1)
  })
})