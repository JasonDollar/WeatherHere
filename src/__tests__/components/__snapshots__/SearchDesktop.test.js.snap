// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDesktop component tests Matches snapshot 1`] = `
<div
  className="Search"
>
  <form
    className="form"
    onSubmit={[MockFunction]}
  >
    <input
      className="input"
      onChange={[MockFunction]}
      placeholder="Szukaj lokalizacji"
      type="text"
      value=""
    />
    <button
      className="button formButton"
      type="submit"
    >
      <span
        className="hidden"
      >
        Szukaj
      </span>
      <Icon
        icon="search"
        title=""
        viewBox="0 0 50 50"
      />
      <span>
        Szukaj
      </span>
    </button>
  </form>
  <button
    className="button buttonGeo"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="hidden"
    >
      Geolokalizacja
    </span>
    <Icon
      icon="gpsArrow"
      title=""
      viewBox="0 0 51.636 51.636"
    />
    <span>
      Geo
    </span>
  </button>
  <button
    className="button buttonSettings"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="hidden"
    >
      Ustawienia
    </span>
    <Icon
      icon="settings"
      title=""
      viewBox="0 0 50 50"
    />
    <span>
      Ustawienia
    </span>
  </button>
</div>
`;
